test/jest/Service.spec.ts

```diff
@@ -90,4 +93,94 @@ describe('Service', () => {
+      });
+      btcClient.walletBalance = jest.fn().mockImplementation(() => {
+        return Promise.resolve({ totalBalance: 10000, confirmedBalance: 10000, unconfirmedBalance: 0 });
+      });
+      components.swapClientManager.swapClients.set('BTC', btcClient);
+
+      const ltcClient = new mockedSwapClient();
+      ltcClient.isConnected = jest.fn().mockImplementation(() => true);
+      ltcClient.channelBalance = jest.fn().mockImplementation(() => {
+        return Promise.resolve({ balance: 0, inactiveBalance: 12345, pendingOpenBalance: 0 });
+      });
+      ltcClient.walletBalance = jest.fn().mockImplementation(() => {
+        return Promise.resolve({ totalBalance: 2000, confirmedBalance: 1500, unconfirmedBalance: 500 });
+      });
+      components.swapClientManager.swapClients.set('LTC', ltcClient);
+
+      const bchClient = new mockedSwapClient();
+      bchClient.isConnected = jest.fn().mockImplementation(() => false);
+      components.swapClientManager.swapClients.set('BCH', bchClient);
+    };
+
+    test('returns balance for all connected clients when currency is not specified', async () => {
+      setup();
+      const result = await service.getBalance({ currency: '' });
+      expect(result.size).toEqual(2);
+
+      const btcBalance = result.get('BTC')!;
+      expect(btcBalance).toBeTruthy();
+      expect(btcBalance.channelBalance).toEqual(70000);
+      expect(btcBalance.pendingChannelBalance).toEqual(190191);
+      expect(btcBalance.inactiveChannelBalance).toEqual(18817);
+      expect(btcBalance.walletBalance).toEqual(10000);
+      expect(btcBalance.unconfirmedWalletBalance).toEqual(0);
+      expect(btcBalance.totalBalance).toEqual(289008);
+
+      const ltcBalance = result.get('LTC')!;
+      expect(ltcBalance).toBeTruthy();
+      expect(ltcBalance.channelBalance).toEqual(0);
+      expect(ltcBalance.pendingChannelBalance).toEqual(0);
+      expect(ltcBalance.inactiveChannelBalance).toEqual(12345);
+      expect(ltcBalance.walletBalance).toEqual(1500);
+      expect(ltcBalance.unconfirmedWalletBalance).toEqual(500);
+      expect(ltcBalance.totalBalance).toEqual(14345);
+    });
+
+    test('returns balance for specified currency', async () => {
+      setup();
+      const result = await service.getBalance({ currency: 'BTC' });
+      expect(result.size).toEqual(1);
+
+      const btcBalance = result.get('BTC')!;
+      expect(btcBalance).toBeTruthy();
+      expect(btcBalance.channelBalance).toEqual(70000);
+      expect(btcBalance.pendingChannelBalance).toEqual(190191);
+      expect(btcBalance.inactiveChannelBalance).toEqual(18817);
+      expect(btcBalance.walletBalance).toEqual(10000);
+      expect(btcBalance.unconfirmedWalletBalance).toEqual(0);
+      expect(btcBalance.totalBalance).toEqual(289008);
+    });
+
+    test('throws in case of invalid currency', async () => {
+      setup();
+      try {
+        await service.getBalance({ currency: 'A' });
```

Comment:
As written I don't think this actually fails if `getBalance` doesn't throw. I believe you should use `await expect(...).rejects.toMatchSnapshot()` here and in the test case below.

Reply:
Indeed, good point!